src/Api/Model/Languageforge/Lexicon/Command/LexUploadCommands.php

```diff
@@ -66,23 +67,52 @@ public static function uploadAudioFile($projectId, $mediaType, $tmpFilePath)
             "audio/x-mpg",
             "audio/x-mpegaudio",
             "audio/x-wav",
-            "audio/wav"
+            "audio/wav",
+            "audio/ogg",
+            "audio/webm",
+            "video/webm"
         );
         $allowedExtensions = array(
             ".mp3",
             ".mpa",
             ".mpg",
             ".m4a",
-            ".wav"
+            ".wav",
+            ".ogg",
+            ".webm"
         );
 
         $response = new UploadResponse();
+
         if (in_array(strtolower($fileType), $allowedTypes) && in_array(strtolower($fileExt), $allowedExtensions)) {
 
             // make the folders if they don't exist
             $project->createAssetsFolders();
             $folderPath = $project->getAudioFolderPath();
 
+            // convert audio file to mp3 or wav format if necessary //LHM
+            // FLEx only supports mp3 or wav format as of 2022-09
+
+            //First, find the duration of the file
+            $ffprobeCommand = 'ffprobe -i ' . $tmpFilePath . ' -show_entries format=duration -v quiet -of csv="p=0" 2> /dev/null';
```

Comment:
@longrunningprocess is right that we should employ some method of sanitizing the variable before passing to the shell.  The variable, while not user input, is coming in from outside the function and so we have less control on its contents.

See https://www.php.net/manual/en/function.escapeshellarg.php

```suggestion
            $ffprobeCommand = 'ffprobe -i ' . escapeshellarg($tmpFilePath) . ' -show_entries format=duration -v quiet -of csv="p=0" 2> /dev/null';
```

Reply:
On second thought, since $tmpFilePath is used often enough below, it would make sense to have a new variable $sanitizedTmpFilePath that you first escape, and then reference in each place below.

Reply:
Got it. This is in the latest commits.